### PR TITLE
Add openAPI specification

### PR DESCRIPTION
**Pull request by iamalexvybornyi**

### CODE REVIEW

File: passport-management-system.yml [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
+        enum:
+          - "Internal"
+          - "International"
+      status:
+        type: "string"
+        description: "Passport status"
+        enum:
+          - "Active"
+          - "Inactive"
+  ApiResponse:
```
Comment (justYP) [inline passport-management-system.yml]: Looks like it is used for error responses only. So I'd suggest another name for this object.

File: passport-management-system.yml [hunk 2]
```diff
@@ -0,0 +1,691 @@
...
+          - "International"
+      person:
+        $ref: "#/definitions/PersonDto"
+      status:
+        type: "string"
+        description: "Passport status"
+        enum:
+          - "Active"
+          - "Inactive"
+  CreatePassportDto:
```
Comment (justYP) [inline passport-management-system.yml]: I'd use `Request`/`Response` instead of `Dto`

File: passport-management-system.yml [hunk 3]
```diff
@@ -0,0 +1,691 @@
...
+        type: "string"
+        description: "Code of the department that issued the passport"
+        example: "111-111"
+      passportType:
+        type: "string"
+        description: "Passport type"
+        enum:
+          - "Internal"
+          - "International"
+      status:
```
Comment (justYP) [inline passport-management-system.yml:573]: is `status` required?

File: passport-management-system.yml [hunk 4]
```diff
@@ -0,0 +1,691 @@
...
+        type: "string"
+        description: "Code of the department that issued the passport"
+        example: "111-111"
+      passportType:
+        type: "string"
+        description: "Passport type"
+        enum:
+          - "Internal"
+          - "International"
+      person:
```
Comment (justYP) [inline passport-management-system.yml:650]: Need to discuss

**Review by justYP — COMMENTED** [review]
Pls make sure the following requirements have covered:

- Get current active passports of a person 
- Search passports by given dates range

File: passport-management-system.yml (hunk 1 above)
Comment (iamalexvybornyi) [reply passport-management-system.yml]: Renamed to ApiError

File: passport-management-system.yml (hunk 2 above)
Comment (iamalexvybornyi) [reply passport-management-system.yml]: Renamed accordingly (CreatePersonDto -> CreatePersonRequest, PersonDto -> PersonResponse, etc.)

File: passport-management-system.yml (hunk 3 above)
Comment (iamalexvybornyi) [reply passport-management-system.yml:573]: Yes, added it to the list of required fields

File: passport-management-system.yml (hunk 4 above)
Comment (iamalexvybornyi) [reply passport-management-system.yml:650]: Sure. I think this exact reference to the person entity will not lead to any issues as this is a PersonDto (now PersonResponse) and it doesn't have any references to passports. But the other entity (Person) has a reference to passports which might lead to the recursion. It seems like I will need to get rid of this reference but let's discuss it in detail later

Comment (iamalexvybornyi) [reply passport-management-system.yml:650]: By the way, it seems like I can remove the "Person" and "Passport" models from the specification as they are not related to the API description, they are only related to the internal implementation details and are not referenced in the requests

**Comment by iamalexvybornyi** [comment]
> Pls make sure the following requirements have covered:
> 
> * Get current active passports of a person
> * Search passports by given dates range

Added the description of the functionality from the first item as I somehow missed it initially. The functionality from the second item has already been described (there are 2 query params in the /passports GET request (**startDate** and **endDate**). If both params are specified the request will return the passports between the given dates

**Pull request merged**